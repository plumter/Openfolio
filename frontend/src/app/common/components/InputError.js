import { Fade } from "@successtar/react-reveal"
import { isValid } from "app/common/util/Helpers";
import SVG from "react-inlinesvg";


const InputError = ({validation, name, text}) => {
    return  <Fade bottom collapse when={!isValid(validation, name)}>
                <div className="text-xs mb-1 pt-2 text-red-500">
                    <SVG
                        src={`/assets/media/svg/info1.svg`}
                        className="stroke-current inline-block h-3.5 mr-1"
                    />
                    {text}
                </div>
            </Fade>

}

export default InputError;