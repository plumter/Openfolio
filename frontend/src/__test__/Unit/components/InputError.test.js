import InputError from 'app/common/components/InputError';
import { create } from 'react-test-renderer';

describe("InputError Component Unit Testing", _ => {

    it('Test InputError Cases',  () => {
      // 1st case
      const component = create(      
          <InputError
              name="email"
              validation={{validated: false, fields:{email: false}}}
              text="Enter a valid email address"
              key={1}
          />,
        );
      expect(component.toJSON()).toMatchSnapshot();

      // 2nd case
      component.update(      
          <InputError
              name="email"
              validation={{validated: true, fields:{email: false}}}
              text="Enter a valid email address"
              key={2}
          />,
        );
      expect(component.toJSON()).toMatchSnapshot();

      // 3rd case
      component.update(      
          <InputError
              name="email"
              validation={{validated: true, fields:{email: true}}}
              text="Enter a valid email address"
              key={3}
          />,
        );
      expect(component.toJSON()).toMatchSnapshot();
    });  
});