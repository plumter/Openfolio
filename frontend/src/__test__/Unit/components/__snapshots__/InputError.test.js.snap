// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputError Component Unit Testing Test InputError Cases 1`] = `
<div
  style={
    Object {
      "height": 0,
      "visibility": "hidden",
    }
  }
>
  <div
    className="react-reveal text-xs mb-1 pt-2 text-red-500"
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    Enter a valid email address
  </div>
</div>
`;

exports[`InputError Component Unit Testing Test InputError Cases 2`] = `
<div
  style={
    Object {
      "height": 0,
      "visibility": undefined,
    }
  }
>
  <div
    className="react-reveal text-xs mb-1 pt-2 text-red-500"
    style={
      Object {
        "opacity": undefined,
      }
    }
  >
    Enter a valid email address
  </div>
</div>
`;

exports[`InputError Component Unit Testing Test InputError Cases 3`] = `
<div
  style={
    Object {
      "height": 0,
      "visibility": "hidden",
    }
  }
>
  <div
    className="react-reveal text-xs mb-1 pt-2 text-red-500"
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    Enter a valid email address
  </div>
</div>
`;
